File: backend/src/TheBlueSky.Flights/Models/FlightSeatStatus.cs
using System.ComponentModel.DataAnnotations;
using TheBlueSky.Flights.Enums;

namespace TheBlueSky.Flights.Models
{
    public class FlightSeatStatus
    {
        public int FlightSeatStatusId { get; set; }

        [Required]
        public int FlightId { get; set; }

        [Required]
        public int AircraftSeatId { get; set; }

        public SeatStatus SeatStatus { get; set; } = SeatStatus.Available;

        public Flight Flight { get; set; } = null!;
        public AircraftSeat AircraftSeat { get; set; } = null!;

    }
}

File: frontend/TheBlueSky.App/src/types/flightSeatStatus.ts
import type { AircraftSeat } from "./aircraftSeat";

export const SeatStatuses = [
  'Available',
  'Hold',
  'Reserved',
  'CheckedIn',
  'Boarded',
  'Blocked',
  'Inoperative',
] as const;

export type SeatStatus = (typeof SeatStatuses)[number];

export type FlightSeatStatus = {
    flightSeatStatusId: number;
    flightId: number;
    seatStatus: SeatStatus;
    aircraftSeat: AircraftSeat;
}